src/Orleans.Core/ApplicationParts/ApplicationPartManagerExtensions.cs

```diff
@@ -59,7 +59,31 @@ public static TFeature CreateAndPopulateFeature<TFeature>(this IApplicationPartM
             applicationPartManager.PopulateFeature(result);
             return result;
         }
-        
+
+        /// <summary>
+        /// Adds the provided assembly to the builder as a framework assembly.
+        /// </summary>
+        /// <param name="manager">The builder.</param>
+        /// <param name="assembly">The assembly.</param>
+        /// <returns>The builder with the additionally added assembly.</returns>
+        public static IApplicationPartManagerWithAssemblies AddFrameworkPart(this IApplicationPartManager manager, Assembly assembly)
+        {
+            if (manager == null)
+            {
+                throw new ArgumentNullException(nameof(manager));
+            }
+
+            if (assembly == null)
+            {
+                throw new ArgumentNullException(nameof(assembly));
+            }
+
+            return new ApplicationPartManagerWithAssemblies(
+                manager.AddApplicationPart(
+                    new AssemblyPart(assembly) {IsFrameworkAssembly = true}),
```

Comment:
What's the difference between the way framework assemblies are treated, vs. non-framework assemblies?

Reply:
Previously we would throw if no non-framework assemblies were added, but now will load default assemblies (everything in the current folder + deps.json). If even one non-framework assembly is added, we will *not* do that (and it's up to the user to specify all of their application parts).